db/00000000000001_initial_schema.up.sql

```diff
@@ -324,15 +296,20 @@ CREATE TYPE BATCH_AGGREGATION_STATE AS ENUM(
-    aggregation_param          BYTEA NOT NULL,                    -- the aggregation parameter (opaque VDAF message)
-    ord                        BIGINT NOT NULL,                   -- the index of this batch aggregation shard, over (task ID, batch_identifier, aggregation_param).
-    state                      BATCH_AGGREGATION_STATE NOT NULL,  -- the current state of this batch aggregation
-    aggregate_share            BYTEA,                             -- the (possibly-incremental) aggregate share; populated unless report_count is 0 or the batch aggregation has been scrubbed.
-    report_count               BIGINT,                            -- the (possibly-incremental) client report count; populated unless the batch aggregation has been scrubbed.
-    checksum                   BYTEA,                             -- the (possibly-incremental) checksum; populated unless the batch aggregation has been scrubbed.
+    id                          BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,  -- artificial ID, internal-only
+    task_id                     BIGINT NOT NULL,                   -- the task ID
+    batch_identifier            BYTEA NOT NULL,                    -- encoded query-type-specific batch identifier (corresponds to identifier in BatchSelector)
+    batch_interval              TSRANGE,                           -- batch interval, as a TSRANGE, populated only for time-interval tasks. (will always match batch_identifier)
+    aggregation_param           BYTEA NOT NULL,                    -- the aggregation parameter (opaque VDAF message)
+    ord                         BIGINT NOT NULL,                   -- the index of this batch aggregation shard, over (task ID, batch_identifier, aggregation_param).
+
+    client_timestamp_interval   TSRANGE NOT NULL,                  -- the minimal interval containing all of client timestamps included in this batch aggregation
+    state                       BATCH_AGGREGATION_STATE NOT NULL,  -- the current state of this batch aggregation
+    aggregate_share             BYTEA,                             -- the (possibly-incremental) aggregate share; populated unless report_count is 0 or the batch aggregation has been scrubbed.
+    report_count                BIGINT,                            -- the (possibly-incremental) client report count; populated unless the batch aggregation has been scrubbed.
+    checksum                    BYTEA,                             -- the (possibly-incremental) checksum; populated unless the batch aggregation has been scrubbed.
+    aggregation_jobs_created    BIGINT,                            -- the number of aggregation jobs that have been created for this batch; populated unless the batch aggregation has been scrubbed.
+    aggregation_jobs_terminated BIGINT,                            -- the number of aggregation jobs that have been terminated (finished or abandoned) for this batch; populated unless the batch aggregation has been scrubbed.
```

Comment:
I think these should be `NULL` to indicate they do not exist, rather than 0/an empty string, which is semantically different (and, at least for the counters, a valid value). This reduces the odds of incorrect interpretations of the data.

Reply:
Good point, I hadn't thought through that 0 is a legitimate, non-null value.